### ISSUE TITLE
Crash with invalid subscription when matching

### ISSUE THREAD

**Issue by crbrox** [telefonicaid/fiware-orion#2257]
The server allows creation of a subscription with an invalid regex and then crashes with segmentation  fault when doing matching .

The invalid regex is in `idPattern`and is `[.*` 
To reproduce
x.sh
```
send() {
  curl -i localhost:1026/v2/$1 -H "Content-type: application/json" --data "$2"
  return $?
}

payload='
{ "subject": { "entities": [ { "idPattern": "[.*", "type": "T" } ], "condition": { "attrs": [ "A" ] } }, "notification": { "http": {"url": "http://localhost:9090/notify"}, "exceptAttrs": [ "X", "Y" ] }, "expires": "2050-04-05T14:00:00.00Z", "status": "active" }
'

payload2='
{ "id": "1", "type": "T", "A": { "value": "aaa" }, "Y": { "value": "yyy" }, "C": { "value": "ccc" }, "X": {"value": "xxx"} }
'
echo
send subscriptions "$payload"  || { echo "Subscription no creada" ; exit -1; }
echo
send entities "$payload2"  || { echo "Entity no creada" ; exit -1; }
echo
```

Core
```
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x00007fc4f1518d32 in __regexec (preg=0xd0c238, string=0x7fc4d4004468 "1", nmatch=0, pmatch=0x0, eflags=0) at regexec.c:248
248	regexec.c: No such file or directory.
(gdb) bt
#0  0x00007fc4f1518d32 in __regexec (preg=0xd0c238, string=0x7fc4d4004468 "1", nmatch=0, pmatch=0x0, eflags=0) at regexec.c:248
#1  0x00000000005c7e7c in EntityInfo::match(std::string const&, std::string const&) ()
#2  0x00000000005c836c in subMatch(CachedSubscription*, char const*, char const*, char const*, char const*, std::vector<std::string, std::allocator<std::string> > const&) ()
#3  0x00000000005cab0e in subCacheMatch(char const*, char const*, char const*, char const*, std::vector<std::string, std::allocator<std::string> > const&, std::vector<CachedSubscription*, std::allocator<CachedSubscription*> >*) ()
#4  0x0000000000628e5e in addTriggeredSubscriptions(std::string, std::string, std::vector<std::string, std::allocator<std::string> > const&, std::map<std::string, TriggeredSubscription*, std::less<std::string>, std::allocator<std::pair<std::string const, TriggeredSubscription*> > >&, std::string&, std::string, std::vector<std::string, std::allocator<std::string> > const&) [clone .isra.192] ()
#5  0x000000000062fe5d in processContextElement(ContextElement*, UpdateContextResponse*, std::string const&, std::string const&, std::vector<std::string, std::allocator<std::string> > const&, std::map<std::string, std::string, std::less<std::string>, std::allocator<std::pair<std::string const, std::string> > >&, std::string const&, std::string const&, std::string const&, Ngsiv2Flavour) ()
#6  0x00000000005e824f in mongoUpdateContext(UpdateContextRequest*, UpdateContextResponse*, std::string const&, std::vector<std::string, std::allocator<std::string> > const&, std::map<std::string, std::string, std::less<std::string>, std::allocator<std::pair<std::string const, std::string> > >&, std::string const&, std::string const&, std::string const&, Ngsiv2Flavour) ()
#7  0x00000000004b2913 in postUpdateContext(ConnectionInfo*, int, std::vector<std::string, std::allocator<std::string> >&, ParseData*, Ngsiv2Flavour) ()
#8  0x00000000004f511f in postEntities(ConnectionInfo*, int, std::vector<std::string, std::allocator<std::string> >&, ParseData*) ()
#9  0x0000000000520001 in restService(ConnectionInfo*, RestService*) ()
#10 0x0000000000516b18 in serve(ConnectionInfo*) ()
#11 0x000000000051a6e1 in connectionTreat(void*, MHD_Connection*, char const*, char const*, char const*, char const*, unsigned long*, void**) ()
#12 0x000000000067b399 in call_connection_handler (connection=connection@entry=0x7fc4d0000ab0) at connection.c:1585
#13 0x000000000067c3c0 in MHD_connection_handle_idle (connection=0x7fc4d0000ab0) at connection.c:2624
#14 0x000000000067f5e4 in MHD_handle_connection (data=0x7fc4d0000ab0) at daemon.c:998
#15 0x00007fc4f24a7182 in start_thread (arg=0x7fc4e6fd2700) at pthread_create.c:312
#16 0x00007fc4f152f47d in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:111
```

Sever started with
` ./contextBroker -logLevel WARNING -fg -httpTimeout 5000 -subCacheIval 600 -notificationMode threadpool:20:3`

**Comment by fgalan** [comment]
Fixed in PR #2260 (release/1.2.0) and PR #2263 (develop)

**Comment by fgalan** [comment]
...and a little follow-up fix in PR #2264 (release/1.2.0) and PR #2265 (develop).
